[v16] WebDiscover: Improve RDS enrollment flow

kimlisa opened the pull request:
backport https://github.com/gravitational/teleport/pull/44671 and https://github.com/gravitational/teleport/pull/45179 to branch/v16

requires https://github.com/gravitational/teleport/pull/45667

manual backport because of very painful storybook difference and design changes

there were so many conflicts that i'll go through the tests again 😢 

Test plan:
- [x] self hosted rds flow single enrollment
- [x] self hosted rds flow auto enrollment (check proper event emit)
- [x] cloud rds flow single enrollment
- [x] cloud rds flow auto enrollment (check proper event emit)
- [x] ec2 ssm self-hosted flow (since the discovery config file got touched)

changelog: Improve web UI enroll RDS flow where VPC, subnets, and security groups are now selectable

kimlisa committed:
Web: Improve RDS enrollment flow (#44671)

- user selects a vpc to see RDS's
- restrict deploying service to user selected vpc
- allow and require selecting subnets and security groups
- refactor EnrollRdsDatabase.tsx into SingleEnrollment and AutoEnrollment
- for self hosted, move configuring discovery config into own step
- fixes a query bug where not all db servers were returning because
  possible duplicates were not accounted for (fixed by removing limit,
  default limit is 1k which should be plenty)

kimlisa committed:
Web: Improve RDS flow 2 (#45179)

* Don't default to auto discover
* Allow redpeploy when auto deploying and improve pending deploy hints
* Allow overwrite existing dbs and skip on heartbeat timeout during enrollment
* Remove allowing custom labels
* Emit rest of discovery config event

ravicious commented:
Should we wait for you to finish testing before approving? What troubles you had with the backport? If it's mostly related to the design components changes, it might be a good idea to forward this feedback to Bartosz.

kimlisa commented:
> Should we wait for you to finish testing before approving? What troubles you had with the backport? If it's mostly related to the design components changes, it might be a good idea to forward this feedback to Bartosz.

nah, i just want to run some tests and compare my PR merged into master side by side to gain my confidence back. most conflicts were easily resolvable (though I'm not sure why it was caused in the first place), it was the story where i had to stop and re-use the old storybook methods

kimlisa committed:
Remove blocking if db_service exists for a vpc

Previously intended to determine if user already auto enrolled,
but now that single and auto enrollment sets the same kind of
labels for db_service, there is no distinction and can let
user be blocked incorrectly.
